src/script/engine.zig

```diff
@@ -309,6 +311,27 @@ pub const Engine = struct {
         try self.stack.pushByteArray(first);
     }
 
+    // OP_2OVER: Copies the pair of items two spaces back in the stack to the front
+    fn op2Over(self: *Engine) !void {
+        const stack_length = self.stack.len();
+        const last_element1 = try self.stack.peek(stack_length - 1);
+        const last_element2 = try self.stack.peek(stack_length - 2);
+        try self.stack.pushByteArray(last_element1);
+        try self.stack.pushByteArray(last_element2);
+    }
+
+    // OP_2SWAP: Swaps the top two pairs of items
+    fn op2Swap(self: *Engine) !void {
+        const a = try self.stack.pop();
+        const b = try self.stack.pop();
+        const c = try self.stack.pop();
+        const d = try self.stack.pop();
+        try self.stack.pushElement(b);
+        try self.stack.pushElement(a);
+        try self.stack.pushElement(d);
+        try self.stack.pushElement(c);
```

Comment:
```
try self.stack.pushElement(second);
try self.stack.pushElement(first);
try self.stack.pushElement(fourth);
try self.stack.pushElement(third);
```

Reply:
or can't you just do
```
if (self.stack.len < 4) { return error.Underflow}
std.mem.swap([]u8, &self.stack.items[4], &self.stack.items[2]);
std.mem.swap([]u8, &self.stack.items[3], &self.stack.items[1]);
```

perf will be way better

Reply:
Or add an
```
fn swap(self: *Stack, idx1: usize, idx2: usize)
```

And call this. I may be reusable in other parts of the code

Reply:
let me check from this

Reply:
@tdelabro can you check from the current implementation plz